src/cpu/valuepred/ValuePredictor.py

```
@@ -0,0 +1,46 @@
+    cxx_header = "cpu/valuepred/valuepred_unit.hh"
+    abstract = True
+
+class EStride(ValuePredictor):
+    type = "EStride"
+    cxx_class = "gem5::valuepred::EStride"
+    cxx_header = "cpu/valuepred/enhanced_stride.hh"
+    abstract = False
+
+    # default params reference 1st cvp paper
+    ways = Param.Int(3, "ways of the EStride")
+    strideWidth = Param.Int(20, "Indicates the number of bits used for stride"
```

Ruff (0.15.2)

[error] `Param` may be undefined, or defined from star imports

(F405)